source/robot_model/include/robot_model/exceptions/CollisionGeometryException.hpp

```diff
@@ -0,0 +1,11 @@
+#include <exception>
+#include <string>
+
+namespace robot_model::exceptions {
+class CollisionGeometryException : public std::invalid_argument {
+
+public:
+    explicit CollisionGeometryException(const std::string& error_message)
+    : invalid_argument("Error in CollisionGeomerty: " + error_message) {}
+};
```

Comment:
since this exception may be thrown in the constructor due to invalid URDF, invalid argument is reasonable. But, it is also thrown in `check_collision` so the specific exception class no longer makes sense. In that case, better to use a base exception class `std::runtime_error`. You also had a typo in the message.

```suggestion
class CollisionGeometryException : public std::runtime_error {

public:
    explicit CollisionGeometryException(const std::string& error_message)
    : runtime_error("Collision geometry error: " + error_message) {}
};
```